week3/src/constants/cardImgList.js

```diff
@@ -0,0 +1,82 @@
+import img_1 from "../assets/img_1.gif";
+import img_2 from "../assets/img_2.gif";
+import img_3 from "../assets/img_3.gif";
+import img_4 from "../assets/img_4.gif";
+import img_5 from "../assets/img_5.gif";
+import img_6 from "../assets/img_6.gif";
+import img_7 from "../assets/img_7.gif";
+import img_8 from "../assets/img_8.gif";
+import img_9 from "../assets/img_9.gif";
+
+const cardImgList = [
+  {
+    name: img_1,
+    alt: "첫 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_2,
+    alt: "두 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_3,
+    alt: "세 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_4,
+    alt: "네 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_5,
+    alt: "다섯 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_6,
+    alt: "여섯 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_7,
+    alt: "일곱 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_8,
+    alt: "여덟 번째 카드",
+    matched: false,
+  },
+  {
+    name: img_9,
+    alt: "아홉 번째 카드",
+    matched: false,
+  },
+];
+
+function mixEasyCardList() {
+  const mixedEasyCardList = cardImgList
+    .sort(() => Math.random() - 0.5)
+    .slice(0, 5);
+  return mixedEasyCardList.concat(mixedEasyCardList);
+}
+
+function mixNormalCardList() {
+  const mixedNormalCardList = cardImgList
+    .sort(() => Math.random() - 0.5)
+    .slice(0, 7);
+  return mixedNormalCardList.concat(mixedNormalCardList);
+}
+
+function mixHardCardList() {
+  const mixedHardCardList = cardImgList.sort(() => Math.random() - 0.5);
+  return mixedHardCardList.concat(mixedHardCardList);
+}
```

Comment:
머ㅓ 이런식으로...?! 아름이가 구현한 방식도 직관적이고 넘 좋은데 난 뭔가 공통된 기능을 수행하는 함수들은 하나로 묶는게 좋을 것 같다구 갠적으로 생각해써 ㅎㅎㅎ 취향차이일수도 ㅎㅎㅎㅎ

```suggestion
function mixCardList(isClick) {
  let cardNum;
  switch (isClick) {
        case 0:
          cardNum = 5;
          break;
        case 1:
          cardNum = 7;
          break;
        case 2:
          cardNum = 9;
          break;
        default:
          break;
      }

  const mixedCardList = cardImgList
    .sort(() => Math.random() - 0.5)
    .slice(0, cardNum);
  return mixedCardList.concat(mixedCardList);
}
```

Reply:
헐 구러네? 난 이 생각은 못했던 것 같아! 위에 switch 문빼고 이렇게 묶으니까 훨씬 기능도 직관적으로 잘 보여서 좋은 것 같아 나두 공통된 기능하는 함수들 하나로 묶으려고 생각 많이 했던 것 같은데 요건 몰랐네..~~ 리팩토링때 참고해야ㅜ지